lib/board.rb

```diff
@@ -0,0 +1,51 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+    puts 
+  end
+
+  def position(input)
+    cells[input.to_i - 1]
+  end
+
+  def full?
+    cells.none? {|cell| cell == " "}
+  end
+
+  def turn_count
+    count = 0
+    cells.each {|cell| 
+      if (cell == "X") || (cell == "O") 
+        count +=1 
+      end
+    }
+    count
+  end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```